docs/source/testing.rst

```diff
@@ -92,46 +102,82 @@ settings.local.yaml:
-```
-======================================================================
-FAIL: test_delete_single_provider_skips_delete_archived_data_if_customer_is_none (api.provider.test.tests_models.ProviderModelTest)
-Assert the delete_archived_data task is not called if Customer is None.
-----------------------------------------------------------------------
-Traceback (most recent call last):
-  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/mock.py", line 1209, in patched
-    return func(*args, **keywargs)
-  File "/Users/nbonilla/Documents/Koku/koku/koku/api/provider/test/tests_models.py", line 43, in test_delete_single_provider_skips_delete_archived_data_if_customer_is_none
-    self.aws_provider.delete()
-AssertionError: no logs of level WARNING or higher triggered on api.provider.models
-
-----------------------------------------------------------------------
-Ran 756 tests in 2025.258s
-FAILED (failures=1)
-```
-
-# Gotchas:
-- The logger is disabled by default during unit tests. If you are building a unit test that asserts that a log occurs, you must re-enable the logger. For example:
-
-```
-import logging
-with self.assertLogs(logger='masu.external.downloader.aws.aws_report_downloader', level='WARN') as cm:
-            logging.disable(logging.NOTSET)
-            self.aws_report_downloader._remove_manifest_file("None")
-            self.assertEqual(['WARN: Could not delete manifest file at'], cm.output)
-```
+
+1. Ensure all migrations are run.
+2. Stop the server `docker-compose stop koku-server`
+3. Run the server with service-ports: :code:`docker-compose run —service-ports koku-server`
+4. set a breakpoint using :code:`import pdb; pdb.set_trace()`
+
+Breakpoints should now be stopped at, in this terminal window.
+
+
+Running PDB in IQE
+------------------
+To run PDB while using dockerized IQE: 
+
+    Start a shell session in the docker container that runs IQE::
+
+        koku/testing/run_test.sh bash
+
+To run a specific test, use the following command. Replace :code:`-k` flag's input to a specific test you want to run::
+
+    iqe tests plugin hccm -k test_api_aws_storage_filtered_top --pdb
+
+optionally you can, run all tests::
+
+    iqe tests plugin hccm --pdb
+
+Any test that fails should start a PDB session.
+
+Smoke testing with IQE
+======================
+
+Prerequisites:
+
+    - koku is deployed via docker-compose or other method, 
+    - you are connected to the Red Hat internal network
+
+Running IQE in Docker
+---------------------
+
+To run IQE Smoke, Vortex or API tests, run one of the following commands, respectively::
+
+    make docker-iqe-smokes-tests
+    make docker-iqe-vortex-tests
+    make docker-iqe-api-tests
+
+Running IQE source code
+-------------------------------
+There need to be clones of the following git repositories:
```

Comment:
It would be helpful to link to the IQE documentation (https://gitlab.cee.redhat.com/insights-qe/hccm-plugin/tree/master) which tells you how to setup IQE & install the hccm-plugin.

Reply:
@abaiken added!